model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2272,50 +2290,29 @@ def _infer_hardware(
         f"Memory calculation result: {min_memory_gb=} for {model_name}, min_kv_cache_size: {min_kv_cache_size}, model_weights_size: {model_weights_size}, is_batch_job: {is_batch_job}"
     )
 
-    if min_memory_gb <= 24:
-        cpus = "10"
-        gpus = 1
-        memory = "24Gi"
-        storage = "80Gi"
-        gpu_type = GpuType.NVIDIA_AMPERE_A10
-    elif min_memory_gb <= 48:
-        cpus = "20"
-        gpus = 2
-        memory = "48Gi"
-        storage = "80Gi"
-        gpu_type = GpuType.NVIDIA_AMPERE_A10
-    elif min_memory_gb <= 96:
-        cpus = "40"
-        gpus = 4
-        memory = "96Gi"
-        storage = "96Gi"
-        gpu_type = GpuType.NVIDIA_AMPERE_A10
-    elif min_memory_gb <= 180:
-        cpus = "20"
-        gpus = 2
-        memory = "160Gi"
-        storage = "160Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
-    elif min_memory_gb <= 320:
-        cpus = "40"
-        gpus = 4
-        memory = "320Gi"
-        storage = "320Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
-    elif min_memory_gb <= 640:
-        cpus = "80"
-        gpus = 8
-        memory = "800Gi"
-        storage = "460Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
-    elif "llama-3-8b-instruct-262k" in model_name:
-        cpus = "20"
-        gpus = 2
-        memory = "40Gi"
-        storage = "40Gi"
-        gpu_type = GpuType.NVIDIA_HOPPER_H100
+    config_map = await _get_recommended_hardware_config_map()
```

Comment:
do we want to have a fallback if this config_map isn't set? it looks like we need to have the helm value `recommendedHardware` set, otherwise I think inferring hardware breaks, and batch completions won't work

Reply:
i can't foresee how we can come up with anything if it's not set. do you mean keep some fallback logics in code?

Reply:
yeah that's one way to go about it, i.e. right now I don't think there's an option to pass hardware choices to batch inference, so we'd have to either let the user provide that info, or just say "you have to set that recommendHardware thing if you want to use batch inference", which might be ok actually

Reply:
i think putting example values in values_sample.yaml should be good enough?

Reply:
yeah that should be fine